### PR TITLE
feat[#314]: Tell user about minimum partition sizes

### LINKED ISSUE #314: Tell user about minimum partition sizes
When selecting manual partitioning, the user has to guess the partition sizes or look for it online. 

It would be preferable to have the manual partitioning page tell the user how large a partition has to be.

### PR DESCRIPTION
**Pull request by muhdsalm**
Closes #314

### CODE REVIEW

File: vanilla_installer/gtk/widget-partition.ui [hunk 1]
```diff
@@ -100,7 +100,7 @@
     <child>
       <object class="AdwPreferencesGroup" id="root_part">
         <property name="title" translatable="yes">Root Partition</property>
-        <property name="description" translatable="yes">The partition to mount under “/”. It will be automatically converted into an LVM thin pool during installation.</property>
+        <property name="description" translatable="yes">The partition to mount under “/”. It will be automatically converted into an LVM thin pool during installation. Minimum space required in the partition is 50GiB.</property>
```
Comment (taukakao) [inline vanilla_installer/gtk/widget-partition.ui:103]: This is not correct, the root needs around 21 GB, though I'm not sure how much exactly.

File: vanilla_installer/gtk/widget-partition.ui [hunk 2]
```diff
@@ -120,7 +120,7 @@
     <child>
       <object class="AdwPreferencesGroup" id="var_part">
         <property name="title" translatable="yes">Var Partition</property>
-        <property name="description" translatable="yes">The partition to mount under “/var”. This directory stores all user data like personal documents and system-wide configuration.</property>
+        <property name="description" translatable="yes">The partition to mount under “/var”. This directory stores all user data like personal documents and system-wide configuration. Minimum space required in the partition is 5GiB.</property>
```
Comment (taukakao) [inline vanilla_installer/gtk/widget-partition.ui:123]: I don't think the installer checks for the size of /var.
5GB is too little tho, should be somewhere around 25-30 GB

**Review by taukakao — CHANGES_REQUESTED** [review]

**Review by Muqtxdir — COMMENTED** [review]
iwould suggest having min requirements specified in `recipe.json` and extracting the values from there. since if we go down the custom image route this would be different with other images

File: vanilla_installer/defaults/disk.py [hunk 3]
```diff
@@ -211,6 +212,7 @@ def __init__(self, parent, partitions, **kwargs):
         super().__init__(**kwargs)
         self.__parent = parent
         self.__partitions = sorted(partitions)
+        self.__recipe = RecipeLoader()
```
Comment (mirkobrombin) [inline vanilla_installer/defaults/disk.py:215]: I suggest to use the instance from the window instead of making a new one

**Review by mirkobrombin — CHANGES_REQUESTED** [review]

**Comment by muhdsalm** [comment]
The code only required /var to have 5GB of space. Should I increase that?

**Review by Muqtxdir — APPROVED** [review]
thanks 👍

**Comment by Muqtxdir** [comment]
> The code only required /var to have 5GB of space. Should I increase that?

now that it's being taken from recipe. we can change it later again in recipe if needed

**Comment by muhdsalm** [comment]
> > The code only required /var to have 5GB of space. Should I increase that?
> 
> now that it's being taken from recipe. we can change it later again in recipe if needed

https://github.com/Vanilla-OS/vanilla-installer/blob/374a1242b1f70498b6bbbd80b0c635f6483d203d/vanilla_installer/defaults/disk.py#L198

**Review by mirkobrombin — COMMENTED** [review]
LGTM ( did not test )

**Review by taukakao — APPROVED** [review]

**Comment by taukakao** [comment]
Thanks for the work. @muhdsalm

**Pull request merged**